What's with all the craze about React? - STEPS 1 and 4

joeyhuaa opened the pull request:
### Reference to issue
- #60

### Description of the changes proposed in the pull request
- I finished writing my blog, including code examples and images
- I was not able to get CodeSandbox working though, not sure why; hence why I excluded Steps 2 and 3 from the title of this PR.

### Reviewers requested:
- @shreyagupta98 
- @atuljayaram

atuljayaram commented on IMC_WhyReact/blog.md:
````diff
@@ -0,0 +1,239 @@
+  document.getElementById('root')
+);
+```
+
````
Very useful code snippets! However, I feel like you just dump them on the reader without giving enough detail about the code snippets can actually be _used_ in an application. Either provide examples and elaborate more in this section, or get rid of this section and integrate these snippets + examples into your two above sections.

joeyhuaa commented:
Hey Atul, thanks for reviewing my work! Really appreciate your honest feedback. I implemented some of your suggestions, but I left the fun pictures and casual tone because I think it makes for a more refreshing read. However, I did explain props and state in more detail and overall, I think my blog comes off as pretty educational when read. 

I also polished all my code examples and added comments to guide readers. Also embedded links to code sandbox so they can experiment on their own. 

Please give this one last look and let me know what you think!

shreyagupta98 commented on IMC_WhyReact/blog.md:
```diff
@@ -0,0 +1,357 @@
+  <img src="./img/0.png" />
+</div>
+
+# What's with all the craze about React?
```
This is great! Just need a few lines of description for each of the coding examples at the end. The code is commented well, but your readers might not look into the code when they first check out your blogs. Should be ready after adding those descriptions!

shreyagupta98 replied:
@joeyhuaa I'm also not able to see the images in the blog itself - can you confirm if this is just on my end?
